dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1730,8 +1736,14 @@ public Map<String, Object> viewTree(User loginUser, long projectCode, long code,
                 // set treeViewDto instances
                 for (int i = limit - 1; i >= 0; i--) {
                     ProcessInstance processInstance = processInstanceList.get(i);
-                    TaskInstance taskInstance = taskInstanceMapper.queryByInstanceIdAndCode(processInstance.getId(),
-                            Long.parseLong(nodeCode));
+                    TaskInstance taskInstance = null;
```

Comment:
```suggestion
TaskInstance taskInstance = taskInstances.stream().filter(instance -> instance.getTaskCode() == Long.parseLong(nodeCode) && instance.getProcessInstanceId() == processInstance.getId()).findFirst().orElse(null);
```

Reply:
For ends the loop early without going through every element

Reply:
> For ends the loop early without going through every element

ok, you are right~